test/extended/util/deployment.go

```diff
@@ -0,0 +1,121 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	o "github.com/onsi/gomega"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// Define test waiting time const
+const (
+	defaultMaxWaitingTime = 300 * time.Second
+	defaultIterationTimes = 30
+)
+
+// Deployment workload definition
+type Deployment struct {
+	Name             string
+	Namespace        string
+	Replicas         string
+	AppLabel         string
+	MaxWaitReadyTime time.Duration
+}
```

Comment:
IMO we should follow pattern in the other files in `util` directory and expose just functions, not structs. And only functions that are really needed for existing tests, like
`WaitForDeploymentReady` and `GetDeploymentResourceVersion` (or template annotations or generation changed)

See `deploymentconfigs.go` for example.

Reply:
Many thanks for the tips and details! Updated.